shell/platform/darwin/ios/framework/Source/flutter_standard_codec_unittest.mm

```diff
@@ -155,12 +178,80 @@
-  FlutterStandardMessageCodec* codec =
-      [FlutterStandardMessageCodec sharedInstance];
-  NSData* encoded = [codec encode:value];
-  FlutterStandardTypedData* decoded = [codec decode:encoded];
-  ASSERT_TRUE(decoded.type == FlutterStandardDataTypeFloat64);
-  ASSERT_TRUE(decoded.elementCount == 2);
-  ASSERT_TRUE(decoded.elementSize == 8);
-  ASSERT_TRUE([data isEqualTo:decoded.data]);
+  checkEncodeDecode(value);
+}
+
+TEST(FlutterStandardCodec, HandlesMethodCallsWithNilArguments) {
+  FlutterStandardMethodCodec* codec =
+      [FlutterStandardMethodCodec sharedInstance];
+  FlutterMethodCall* call =
+      [FlutterMethodCall methodCallWithMethodName:@"hello" arguments:nil];
+  NSData* encoded = [codec encodeMethodCall:call];
+  FlutterMethodCall* decoded = [codec decodeMethodCall:encoded];
+  ASSERT_TRUE([decoded isEqualToMethodCall:call]);
+}
+
+TEST(FlutterStandardCodec, HandlesMethodCallsWithSingleArgument) {
+  FlutterStandardMethodCodec* codec =
+      [FlutterStandardMethodCodec sharedInstance];
+  FlutterMethodCall* call =
+      [FlutterMethodCall methodCallWithMethodName:@"hello" arguments:@42];
+  NSData* encoded = [codec encodeMethodCall:call];
+  FlutterMethodCall* decoded = [codec decodeMethodCall:encoded];
+  ASSERT_TRUE([decoded isEqualToMethodCall:call]);
+}
+
+TEST(FlutterStandardCodec, HandlesMethodCallsWithArgumentList) {
+  FlutterStandardMethodCodec* codec =
+      [FlutterStandardMethodCodec sharedInstance];
+  NSArray* arguments = @[ @42, @"world" ];
+  FlutterMethodCall* call =
+      [FlutterMethodCall methodCallWithMethodName:@"hello" arguments:arguments];
+  NSData* encoded = [codec encodeMethodCall:call];
+  FlutterMethodCall* decoded = [codec decodeMethodCall:encoded];
+  ASSERT_TRUE([decoded isEqualToMethodCall:call]);
+}
+
+TEST(FlutterStandardCodec, HandlesSuccessEnvelopesWithNilResult) {
+  FlutterStandardMethodCodec* codec =
+      [FlutterStandardMethodCodec sharedInstance];
+  NSData* encoded = [codec encodeSuccessEnvelope:nil];
+  FlutterError* error = nil;
+  id decoded = [codec decodeEnvelope:encoded error:&error];
+  ASSERT_EQ(error, nil);
+  ASSERT_EQ(decoded, nil);
+}
+
+TEST(FlutterStandardCodec, HandlesSuccessEnvelopesWithSingleResult) {
+  FlutterStandardMethodCodec* codec =
+      [FlutterStandardMethodCodec sharedInstance];
+  NSData* encoded = [codec encodeSuccessEnvelope:@42];
+  FlutterError* decodedError = nil;
+  id decodedResult = [codec decodeEnvelope:encoded error:&decodedError];
+  ASSERT_EQ(decodedError, nil);
+  ASSERT_EQ(decodedResult, @42);
+}
+
+TEST(FlutterStandardCodec, HandlesSuccessEnvelopesWithResultMap) {
+  FlutterStandardMethodCodec* codec =
+      [FlutterStandardMethodCodec sharedInstance];
+  NSDictionary* result = @{ @"a" : @42, @42 : @"a" };
+  NSData* encoded = [codec encodeSuccessEnvelope:result];
+  FlutterError* decodedError = nil;
+  id decodedResult = [codec decodeEnvelope:encoded error:&decodedError];
+  ASSERT_TRUE([decodedResult isEqualTo:result]);
```

Comment:
NSObjects actually have an [isEqualTo:](https://developer.apple.com/reference/objectivec/nsobject/1393823-isequalto?language=objc) method as part of the `NSComparsonMethods` protocol. Nice trap for the unwary.